### PR TITLE
Add file field support

### PR DESCRIPTION
**Pull request by bcvandendool**
Ive added support for clearable file field widget, and updated the file field test case to pass with the new setup. It is mostly a port from the bootstrap4 code, i believe i have changed everything to the bootstrap 5 equivalent, but i might have missed something.

### CODE REVIEW

**Comment by smithdc1** [comment]
Thanks for the PR. 👍

Could I ask you to have a look at how the tests were written for https://github.com/django-crispy-forms/crispy-bootstrap5/pull/44? 

We're trying to improve the tests by asserting the whole field rather than html snippets. It will also make it far easier to review. Could you have a look at that?

**Comment by bcvandendool** [comment]
Ive changed the tests to be similar to those.

**Comment by bcvandendool** [comment]
Ive fixed the formatting issue, sorry about that.

**Comment by smithdc1** [comment]
No problem. I hadn't realised until now that I hadn't setup the tests to run from forks. 🕵️

I've had a quick glance at your patch this morning and it looks great. I just want to have a bit of a play as there's a few different scenarios here.

**Comment by smithdc1** [comment]
Sorry for the incremental review, trying to make progress where I can. 

I'm seeing the help text rendering twice. Here's an image and here's a demo bootstrap5 test site [here](https://github.com/smithdc1/crispy-test-project/tree/bs5) showing the error. 

I think it's coming from the help text being rendered both in the `field.html` and `field_file.html`, but don't have time to look more closely right now. 

image

**Comment by bcvandendool** [comment]
You were right, it was indeed added in both `field.html` and `field_file.html`, ive fixed it by removing it from `field_file.html` as there is no need for special handling of the help text in this case as far as i understand.

File: tests/results/test_clearable_file_field.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                            <input type="checkbox" name="clearable_file-clear" id="clearable_file-clear_id" class="form-check-input" >
+                            <label class="form-check-label mb-0" for="clearable_file-clear_id">Clear</label>
+                        </span>
+                    </span>
+                </div>
+            </div>
+            <div class="input-group mb-0 ">
+                <input type="file" name="clearable_file" class="form-control" id="id_clearable_file">
+            </div>
+            <div class="input-group mb-0">
```
Comment (smithdc1) [inline tests/results/test_clearable_file_field.html:22]: This line here looks odd, I don't think think we need it? I'm not sure where it is coming from? 🤔

Comment (bcvandendool) [reply tests/results/test_clearable_file_field.html]: Fixed, it was a remnant of the double help text.

File: crispy_bootstrap5/templates/bootstrap5/layout/field_file.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            <span class="form-check">
+                <input type="checkbox" name="{{ widget.data.checkbox_name }}" id="{{ widget.data.checkbox_id }}" class="form-check-input"{% if field.field.disabled %} disabled{% endif %} >
+                <label class="form-check-label mb-0" for="{{ widget.data.checkbox_id }}">{{ widget.data.clear_checkbox_label }}</label>
+            </span>
+        </span>
+        {% endif %}
+    </div>
+</div>
+{% endif %}
+<div class="input-group mb-0 {% if field.errors %} is-invalid{%endif%}">
```
Comment (smithdc1) [inline crispy_bootstrap5/templates/bootstrap5/layout/field_file.html:23]: I don't think we need the `mb-0` here?

**Comment by smithdc1** [comment]
Thanks for the updates here. 

I'm not seeing the error message when the field fails validation. Appologies for the upcoming blobs of html. I think it would be useful to also add a test for the failing version of this field? 

image

We're currently getting
``` html
<div id="div_id_default_form_failing-file_field" class="mb-3">
   <label for="id_default_form_failing-file_field" class="form-label requiredField">
   File field<span class="asteriskField">*</span> </label> 
   <div class=" mb-2">
      <div class="input-group mb-0  is-invalid"> <input type="file" name="default_form_failing-file_field" class="form-control is-invalid" id="id_default_form_failing-file_field" required=""></div>
   </div>
   <span id="error_1_id_default_form_failing-file_field" class="invalid-feedback"><strong>This field is required.</strong></span> 
</div>
```

I think we need
``` html

<div id="div_id_default_form_failing-file_field" class="mb-3">
   <label for="id_default_form_failing-file_field" class="form-label requiredField">
   File field<span class="asteriskField">*</span> </label>
   <div class=" mb-2">
      <div class="input-group mb-0  is-invalid"> <input type="file" name="default_form_failing-file_field" class="form-control is-invalid" id="id_default_form_failing-file_field" required=""></div>
      <span id="error_1_id_default_form_failing-file_field" class="invalid-feedback"><strong>This field is required.</strong></span>
   </div>
</div>
```

**Comment by bcvandendool** [comment]
Ive now moved the help and error handling to the file_field.html, im not sure if the way i handled it in field.html is the cleanest though. Ive also added tests for failing validation of the file fields.

**Comment by smithdc1** [comment]
This looks great. Thank you very much!

**Pull request merged**